hyperspy/model.py

```diff
@@ -1257,11 +790,10 @@ def fit(self, fitter=None, method='ls', grad=False,
         elif fitter == "odr":
             modelo = odr.Model(fcn=self._function4odr,
                                fjacb=odr_jacobian)
-            mydata = odr.RealData(
-                self.axis.axis[
-                    self.channel_switches], self.spectrum()[
-                    self.channel_switches], sx=None, sy=(
-                    1 / weights if weights is not None else None))
+            mydata = odr.RealData(self.axis.axis[self.channel_switches],
```

Comment:
`Model2D` does not have `self.axis` or `self._function4odr` for the moment. I'm not sure if the fitter supports 2d data, or we'll have to trick it.. Any way, another note for the "cleanup" once this is merged - either we'll have to add all the missing functions and change this bit to actually work, or raise an error before it gets to here.